code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/contentprovider/AepUIContentProvider.kt

```diff
@@ -0,0 +1,34 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.aepcomposeui.contentprovider
+
+import com.adobe.marketing.mobile.aepuitemplates.AepUITemplate
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * Represents a strategy to provide content for rendering the UI.
+ * Allows components like extensions which provide content to expose an implementation
+ * for the app to use.
```

Comment:
Not sure if we should describe details of how extensions will consume it in public docs. Can we keep it simple `Responsible for retrieving and refreshing data as required by the UI`